### PR TITLE
pillow-dependency update

### LINKED ISSUE #10633: Type Error when saving jpg image (Pillow 3)
### Bug report

**Bug summary**

Saving a jpg file via `plt.savefig("plot.jpg")` fails with a type error in matplotlib 2.2.0rc1. Did the required pillow version change? The docs say it's 2.0 or higher.

**Code for reproduction**

```
import matplotlib
matplotlib.use("Agg")
import matplotlib.pyplot as plt

plt.plot([1,3,1])
plt.savefig("plot.jpg")
```

**Actual outcome**

```
Traceback (most recent call last):
  File "D:\*******\GI_savefigjpg.py", line 16, in <module>
    plt.savefig("plot.jpg")
  File "c:\******\python-2.7.10.amd64\lib\site-packages\matplotlib\pyplot.py", line 710, in savefig
    res = fig.savefig(*args, **kwargs)
  File "c:\******\python-2.7.10.amd64\lib\site-packages\matplotlib\figure.py", line 2033, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "c:\******\python-2.7.10.amd64\lib\site-packages\matplotlib\backend_bases.py", line 2261, in print_figure
    **kwargs)
  File "c:\******\python-2.7.10.amd64\lib\site-packages\matplotlib\backends\backend_agg.py", line 582, in print_jpg
    return background.save(filename_or_obj, format='jpeg', **options)
  File "c:\******\python-2.7.10.amd64\lib\site-packages\PIL\Image.py", line 1665, in save
    save_handler(self, fp, filename)
  File "c:\******\python-2.7.10.amd64\lib\site-packages\PIL\JpegImagePlugin.py", line 691, in _save
    ImageFile._save(im, fp, [("jpeg", (0, 0)+im.size, 0, rawmode)], bufsize)
  File "c:\******\python-2.7.10.amd64\lib\site-packages\PIL\ImageFile.py", line 484, in _save
    e = Image._getencoder(im.mode, e, a, im.encoderconfig)
  File "c:\******\python-2.7.10.amd64\lib\site-packages\PIL\Image.py", line 430, in _getencoder
    return encoder(mode, *args + extra)
TypeError: integer argument expected, got float

```

**Expected outcome**

No error should occur.

**Matplotlib version**
  * Operating system: **Windows 8.1**
  * Matplotlib version: **2.2.0rc1**
  * Matplotlib backend: **Agg**, also tested with `%matplotlib inline`
  * Python version: **2.7.10**
  * Other libraries:
    ```
    import PIL
    print PIL.VERSION          # prints 1.1.7
    print PIL.PILLOW_VERSION   # prints 3.0.0
    ```

### PR DESCRIPTION
**Pull request by ImportanceOfBeingErnest**
## PR Summary

Document the minimum optional Pillow version.  
Fixes #10633. 

## PR Checklist

- [x] Code is PEP 8 compliant
- [x] Documentation is sphinx and numpydoc compliant

### CODE REVIEW

**Comment by dstansby** [comment]
It really annoys me that the string "3.4" is hardcoded in several places in the code, instead of being in one place that it is pulled from, but I guess there's no easy way to do that?

**Comment by ImportanceOfBeingErnest** [comment]
This PR changes the code to have that number appear only once in the code, namely in backend_bases.py. Any descendent backend can import `_has_pil`. 

Or are you saying the documentation should pull it from the code?

**Comment by dstansby** [comment]
Yeah, ideally. Presumably it's also defined in a pip requirements file somewhere too.

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -55,9 +55,12 @@
 
 try:
     from PIL import Image
+    from PIL import PILLOW_VERSION
+    from distutils.version import LooseVersion
+    assert LooseVersion(PILLOW_VERSION) >= LooseVersion("3.4")
     _has_pil = True
     del Image
-except ImportError:
+except (ImportError, AssertionError):
```
Comment (dstansby) [inline lib/matplotlib/backend_bases.py]: Instead of catching an `AssertionError`, I think it would be better to have an if block that checks the version.

**Comment by ImportanceOfBeingErnest** [comment]
> Yeah, ideally. 

I'd agree, but I don't know how that would be done with Sphinx. 

> Presumably it's also defined in a pip requirements file somewhere too.

I don't think so because Pillow is not actually a requirement, just an optional thing, which, if present, allows to have further functionality included.

File: lib/matplotlib/backend_bases.py (hunk 1 above)
Comment (ImportanceOfBeingErnest) [reply lib/matplotlib/backend_bases.py]: Ok, changed that now. Any reason for this? Is it bad style or does it have any side effects?

**Comment by timhoffm** [comment]
Don‘t know either how it could be done with Sphinx. But since we‘ll rarely change the version dependency, it’s ok to hard code in the docs.

**Pull request merged**